pkgs/build-support/make-darwin-app/default.nix

```diff
@@ -0,0 +1,74 @@
+{
```

Comment:
Maybe we should call this unpack-darwin-app or something? The current name causes a bit of confusion with make-darwin-bundle.

Reply:
Renamed to `mkDarwinAppFromBin`. I think that conveys better what's going on.

Decided against unpack as that seems like something exclusively for `unpackPhase`. Also considered `mkDarwinDerivation`, but that's too generic imo.

Also added a description of the function.

Reply:
Naming things is hard :smile: I'll go with `extract-darwin-app`.

Reply:
That's done now. Thanks for the suggestions!